pkg/scheduler/scheduler.go

```diff
@@ -319,6 +319,7 @@ func (sched *Scheduler) recordSchedulingFailure(fwk framework.Framework, podInfo
 		Reason:  reason,
 		Message: err.Error(),
 	}, nominatedNode); err != nil {
+		sched.SchedulingQueue.DeleteNominatedPodIfExists(podInfo.Pod)
```

Comment:
Could you pls add a comments to explain in which case the pod is still in the queue? This doesn't seems like a normal case.

Reply:
It's just a rollback operator of `AddNominatedPod`. 
When we preempted successfully, we'll add the `nominatedNode` to the SchedulingQueue by calling `AddNominatedPod()`, after then we will update pod's status, but if some error happens here, the pod's status would be dis-match with the cache. 
Although in the next preemption cycle, we may remove the cache by calling the `AddNominatedPod`,  but I think add a revert operator here is better.